snapmaker/src/service/quick_stop.cpp

```diff
@@ -247,17 +248,24 @@ void QuickStopService::Park() {
     break;
 
   case MODULE_TOOLHEAD_CNC:
-    if (current_position[Z_AXIS] + CNC_SAFE_HIGH_DIFF > Z_MAX_POS) {
+    if (source_ == QS_SOURCE_POWER_LOSS) {
+
+    }
+    else {
+      Z_enable;
```

Comment:
no need to enable Z axis here, because you didn't disable Z axis when triggered by PAUSE event